src/snowflake/snowpark/modin/plugin/extensions/datetime_index.py

```diff
@@ -167,7 +167,11 @@ def __init__(
             "copy": copy,
             "name": name,
         }
-        self._init_index(data, _CONSTRUCTOR_DEFAULTS, query_compiler, **kwargs)
+        self._query_compiler = DatetimeIndex.get_query_compiler(
+            data, _CONSTRUCTOR_DEFAULTS, query_compiler, **kwargs
+        )
+        # `_parent` keeps track of any Series or DataFrame that this Index is a part of.
+        self._parent = None
```

Comment:
this should be None by default right?

Reply:
it's just we have to create `_parent` property in this Index object. Otherwise index._parent will throw `AttributeError: 'Index' object has no attribute '_parent'`